cli/src/main/java/com/box/l10n/mojito/cli/filefinder/FileFinder.java

```diff
@@ -193,7 +217,12 @@ private List<FileMatch> getSourceMatcheCandidates(FileType fileType) throws File
     }
 
     SourceFileVisitor sourceFileVisitor =
-        new SourceFileVisitor(fileType, sourceDirectory, sourcePathFilterRegex);
+        new SourceFileVisitor(
+            fileType,
```

Comment:
Do we walk the tree for each and every different fileType? vs.: instead of doing a single walk of the tree - and checking every file against one or more file type extensions.

Reply:
We walk the tree once (in fact we walk it twice: one for the source file and one for target files) and check all filetypes registered or provided as command options. Trying to remove the target scan could make sense since the limiting factor is IO. That said checking for all file types become expansive too and it doesn't seem to be a really valid use case after all. We use a file type in all of our usages. It feels like it could/should be made a mandatory option